_posts/2020-12-21--kinesis-streams-part-2.md

```diff
@@ -0,0 +1,591 @@
+
+
+```
+aws kinesis register-stream-consumer --consumer-name my_highway \
+      --stream-arn arn:aws:kinesis:<region>:<account_id>:stream/the-best-stream
+```
+
+Though apparently, there's no way to register an enhanced consumer through the _AWS Kinesis_ console yet (if you find one, let me know!), you can list and delete the existing ones:
+
+
+![consumer_console](/img/kinesis/consumer_console.png){: .img.centered }
+
+ 
+
+In the case of _AWS Lambda_, you can now use this enhanced consumer when configuring the Kinesis trigger for your function. Note, that you can only have **a single Lambda function** using a given enhanced consumer. You will need to create a separate enhanced stream consumer for each of your Lambda functions.
+
+
+
+### Which one to chose?
+
+Though the dedicated throughput sounds great on its own, this is not the only perk you are getting with enhanced fan-out. Together with it comes even less **read propagation latency**, aka the time between when the data is written to the stream and when it is consumed. This is in part due to the dedicated throughput “highway" each consumer is getting, and in part to the fact that enhanced fan-out uses **HTTP2** instead of HTTP, allowing to use persistent connections and to **push** records to the consumers, instead of consumers **polling** the stream.
+
+When using enhanced fan-out, you will have records available to be consumed about _70ms_ after you wrote them to the stream. In comparison, it takes about _200ms_ for the shared throughput consumers and this latency increases with each additional consumer that shares the throughput. Why is that? You will have to keep reading to find out 😉 
+
+By now, you might be thinking, why would anyone use the standard shared throughput at all? One important reason is this: while the standard shared throughput consumers normally read the data **for free**, you will have to [pay](https://aws.amazon.com/kinesis/data-streams/pricing/) for having all the perks of enhanced fan-out. You will pay for both, the **amount of data** consumed, as well as for the **number of consumers per shard**.
+
+So, your highway turns out to be a toll road.
+
+
+### Too many consumers are barely enough
+
+Lastly, you don’t need to choose one way of consuming the stream over the other. The same way you can mix and match different types of consumers (direct service integrations with _Kinesis Firehose_ and _Analytics_, Lambda functions, etc.), you can also use **both**, shared throughput and enhanced fan-out with the same Kinesis stream. This will allow you to have up to **20 consumers with dedicated throughput, as well as 2-3 consumers that will share the 2 MB throughput**.
```

Comment:
Lastly -> Finally

I'd drop the comma: "you can also use **both** shared throughput"...